updating so that we continue to use ikm.dev module naming and wrap caching-api are used

jlgrock opened the pull request:
note that mostly this is moving the module names back, adding the caching-api (and undoing the exclude).  While I was in there, I blew away the Jenkins files (unneeded) and refactored the gitignore files.  Please take a look and you can see what we want to do to use this.  I'll continue updating/testing, but wanted to get this to you quickly so you could see the desired state.

jlgrock committed:
updating so that we continue to use ikm.dev module naming and wrap caching-api lib that are used

besidev commented on commons-io/pom.xml:
```diff
@@ -140,7 +140,7 @@
 									</artifact>
 									<moduleInfoSource>
-										open module org.apache.commons.io {
+										open module dev.ikm.jpms.commons.io {
```
Shouldn't the module name remain `org.apache.commons.io`? Generally, only the artifact `groupId` is changed to `dev.ikm.jpms` to reference the JPMS compliant version.

jlgrock replied:
so you will encounter module runtime collisions if you do this, when in theory, they could be both loaded (though it is not suggested either).  In general, much like maven artifact groups, best practice is to start it with your domain if you are the owner.  Similarly, we will not and cannot publish to maven central under `org.apache` because we do not own this domain.

indritbeqiri replied:
Thank you for the clarification, @jlgrock. We now understand why you’ve chosen to use your own domain for both the groupId and the module name when repackaging or distributing libraries.

besidev commented on commons-lang3/pom.xml:
```diff
@@ -140,7 +140,7 @@
 									</artifact>
 									<moduleInfoSource>
-										open module org.apache.commons.lang3 {
+										open module dev.ikm.jpms.commons.lang3 {
```
Again, shouldn’t the module name remain `org.apache.commons.lang3`? Although it’s not mandatory, it would be consistent with the naming structure of the other modules in the JPMS repository, which generally matches the automatic module naming convention. A very good example is [Eclipse Collections](https://github.com/ikmdev/jpms-deps/blob/5538fefda73d454bdb0570714f1c0a8c14930581/eclipse-collections/pom.xml#L232)

jlgrock replied:
yeah, you found another bug in our releases.  Someone didn't understand how to publish eclipse collections.  That is a defect we will correct.

besidev commented on commons-text/pom.xml:
```diff
@@ -152,8 +152,8 @@
 									<moduleInfoSource>
-										open module org.apache.commons.text {
-											requires org.apache.commons.lang3;
+										open module dev.ikm.jpms.commons.text {
+											requires dev.ikm.jpms.commons.lang3;
```
Same as before, the name of the module has been changed to `dev.ikm.jpms.commons.text`. Is this just an arbitrary convention for Apache Commons libs not following the auto naming convention which is `org.apache.commons.text`? Is there any specific reason?

jlgrock replied:
see above - same concept

besidev reviewed (commented):
Generally the artifact `groupId` should change to reference the JPMS compliant module, while the module names should continue to follow the automatic module naming values.

indritbeqiri commented:
Please @jlgrock , can you update:
1. `pom.xml` of `commons-text`
https://github.com/Sandec/jpms-deps/blob/19c58ba36571446b3a5294fa96060220c9651465/commons-text/pom.xml

Line:29
<apache-commons.version>3.17.0-r1</apache-commons.version>

2. `pom.xml` of `hapi-fhir-r4`
https://github.com/Sandec/jpms-deps/blob/19c58ba36571446b3a5294fa96060220c9651465/hapi-fhir-r4/pom.xml

Line:30
<hapi-fhir-utilities.version>6.3.3-r8</hapi-fhir-utilities.version>

Line:163
requires dev.ikm.jpms.commons.lang3;

3. `pom.xml` of `hapi-fhir-structures-r4`
https://github.com/Sandec/jpms-deps/blob/19c58ba36571446b3a5294fa96060220c9651465/hapi-fhir-structures-r4/pom.xml

Lines: 29-30
<hapi-fhir-utilities.version>6.3.3-r8</hapi-fhir-utilities.version>
<hapi-fhir-r4.version>6.3.3-r9</hapi-fhir-r4.version>

Line: 179
requires dev.ikm.jpms.org.hl7.fhir.r4;

jlgrock committed:
jpro team found a couple of issues with the code, correcting.

jlgrock commented:
Sorry about the slow turnaround.  I corrected, though the last one (179) was a different mistake, because the org was still in there.  I have adjusted in the h4 pom

indritbeqiri committed:
Fix module name for `commons-lang3` artifact

indritbeqiri committed:
Align `dev.ikm.jpms.org.hl7.fhir.r4` module name with `tinkar-core` requirements

indritbeqiri committed:
Align `dev.ikm.jpms.hapi.fhir.structures.r4` module name with tinkar-core requirements and some code cleanup

indritbeqiri commented:
In recent commits, the module names `dev.ikm.jpms.org.hl7.fhir.r4` and `dev.ikm.jpms.org.hl7.fhir.utilities` have been aligned with those used in`tinkar-core`.

Here is an example of a `module-info.java` class from `tinkar-core`:
<img width="568" alt="Screenshot 2024-09-18 at 7 21 41 PM" src="https://github.com/user-attachments/assets/bc3a312b-f565-439b-893e-3a3d9193f7e7">

It appears that the same naming structure of the original artifacts has been maintained:
<img width="380" alt="Screenshot 2024-09-18 at 7 29 04 PM" src="https://github.com/user-attachments/assets/b8e41e7e-d0f0-41ca-9a28-f00d76170815">

All tests have passed, and the changes are ready to be merged.

besidev commented:
I have just tested everything on my system too and I confirm all works now. We can proceed to merge at this point 🙂.
Thank you Justin.
